### PR TITLE
 #2288: add metadata to StreamStorage ledgers

### PR DESCRIPTION
**Pull request by dmercuriali**
Descriptions of the changes in this PR:

Added 'application' and 'component' metadata to StreamStorage ledgers.
application: bk-stream-storage-service
component: state-store / checkpoint-store

Master Issue: #2288 

> - [X] [skip bookkeeper-server bookie tests]: skip testing `org.apache.bookkeeper.bookie` in bookkeeper-server module.
> - [X] [skip bookkeeper-server client tests]: skip testing `org.apache.bookkeeper.client` in bookkeeper-server module.
> - [X] [skip bookkeeper-server replication tests]: skip testing `org.apache.bookkeeper.replication` in bookkeeper-server module.
> - [X] [skip bookkeeper-server tls tests]: skip testing `org.apache.bookkeeper.tls` in bookkeeper-server module.
> - [X] [skip bookkeeper-server remaining tests]: skip testing all other tests in bookkeeper-server module.
> - [ ] [skip integration tests]: skip docker based integration tests. if you make java code changes, you shouldn't skip integration tests.
> - [ ] [skip build java8]: skip build on java8. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> - [ ] [skip build java11]: skip build on java11. *ONLY* skip this when *ONLY* changing files under documentation under `site`.

### CODE REVIEW

File: stream/statelib/src/main/java/org/apache/bookkeeper/statelib/impl/rocksdb/checkpoint/dlog/DLCheckpointStore.java [hunk 1]
```diff
@@ -98,6 +100,9 @@ public OutputStream openOutputStream(String filePath) throws IOException {
         try {
             DistributedLogManager dlm = namespace.openLog(
                 filePath);
+            LedgerMetadata metadata = new LedgerMetadata();
+            metadata.setApplication(Constants.LEDGER_METADATA_APPLICATION_STREAM_STORAGE);
+            metadata.setComponent("checkpoint-store");
```
Comment (eolivelli) [inline stream/statelib/src/main/java/org/apache/bookkeeper/statelib/impl/rocksdb/checkpoint/dlog/DLCheckpointStore.java]: this looks like a dead store,
you are not using this variable

**Comment by eolivelli** [comment]
@mino181295 PTAL

File: stream/statelib/src/main/java/org/apache/bookkeeper/statelib/impl/rocksdb/checkpoint/dlog/DLCheckpointStore.java (hunk 1 above)
Comment (dmercuriali) [reply stream/statelib/src/main/java/org/apache/bookkeeper/statelib/impl/rocksdb/checkpoint/dlog/DLCheckpointStore.java]: passed to the openAsyncLogWriter call

Comment (eolivelli) [reply stream/statelib/src/main/java/org/apache/bookkeeper/statelib/impl/rocksdb/checkpoint/dlog/DLCheckpointStore.java]: ok

**Comment by eolivelli** [comment]
@sijie all tests are passing.
can you please take a look ?

**Comment by mino181295** [comment]
Looks good to me

**Pull request merged**